honeybee_energy/properties.py

```diff
@@ -1,18 +1,20 @@
 """Energy Properties."""
-from .boundarycondition import boundary_conditions
 from .construction import Construction
-from .lib.constructionset import generic as generic_constructionset
 
 
 class EnergyProperties(object):
     """Energy Properties for Honeybee geometry."""
 
-    def __init__(self, face_type):
-        self.face_type = face_type
-        self.boundary_condition = boundary_conditions.outdoors
-        # this will be set by user
-        self._construction = None
-        # this will be set by parent zone
+    __slots__ = ('_face_type', '_boundary_condition',
+                 '_construction', '_constructionset')
+
+    def __init__(self, face_type, boundary_condition, construction=None):
+        # private properties to set construction from construction-set.
+        self._face_type = face_type
+        self._boundary_condition = boundary_condition
+        # construction set by user
+        self.construction = construction
+        # Constructionset can only be be set from the parent zone
         self._constructionset = None
```

Comment:
I agree that this is the right approach but I just wanted to hear your thoughts about whether this `_constructionset` will be an id or an actual `ConstructionSet` object.  I feel that it could be an object instance and we would still be ok in Grasshopper as long as we add a `__copy__` or `duplicate` method to `EnergyProperties` that doesn't make a copy of the whole `ConstructionSet` instance.  We can have something similar on the zone level and, as long as we make the `ConstructionSet` objects immutable, this should all be good. What do you think?

Reply:
I haven't figured that out yet as it depends on how we will handle constructions/ materials. It can be an id if we can retrieve it from a database or what you mentioned above. We will know the best solution once we have the full workflow. I would say for now your solution makes a lot of sense.